[MHV-52918] Added V3 va-pagination style attribute + unit test

vsaleem opened the pull request:
## Summary

- _Updated VaPagination component to use V3 style attribute provided by Design.va.gov system_
- _User is now able to navigate to beginning of thread if they are currently on the last page or middle to end pages_

## Related issue(s)

### [MHV-45978](https://jira.devops.va.gov/browse/MHV-45978) - Pagination on last page of thread list view

### [MHV-52918](https://jira.devops.va.gov/browse/MHV-52918) - Implement the V3 pagination component
###

## Testing done

- Unit test checks for styling attribute provided by Design.va.gov

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/a446acc6-1b01-430b-8aad-0142fdf8d896)|![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/255ce61d-64cf-4664-b341-52a0f22506b8)| 
| Desktop |![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/71f7a9d3-a1e7-40b7-beef-661e14dfde6d)![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/2ee96031-6b50-414b-824e-0205b75fbe63) |![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/76fec89b-98c2-4e3f-931d-6ec7f389b0cc)![image](https://github.com/department-of-veterans-affairs/vets-website/assets/59583325/82a3ba04-58fc-40ee-9111-c424d9fb9b5c)|

## What areas of the site does it impact?

*Va.gov - Secure Messaging*

## Acceptance criteria

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

Commit:
[MHV-52918] Added V3 va-pagination style attribute + unit test

oleksii-morgun-va commented on src/applications/mhv/secure-messaging/components/ThreadList/ThreadsList.jsx:
```diff
@@ -17,7 +17,7 @@ const ThreadsList = props => {
   } = props;
 
-  const MAX_PAGE_LIST_LENGTH = 5;
+  const MAX_PAGE_LIST_LENGTH = 7;
```
I am wondering if increasing this may cause issues at high level zoom?

vsaleem replied:
The numbers does indeed cut off a bit in super zoom... made the change back to 5.👍

vsaleem replied:
Received feedback from A11y -- the recommendation is to set the length to 7 so that the middle numbers are displayed in the pagination component and note that zoom capability goes up to 400%.

alex-morgun-1 reviewed (changes requested):
I like this new pagination component, far more user friendly. I left a question on increasing the max number of pages. Also  see the defect with the pagination disappearing on the last page with 1 thread may still need to be looked at

fazilqa committed:
fix: fix pagination test

Commit:
[MHV-52918] bug fix; displays pagination on last page if threadList has length

fazilqa committed:
fix: fix trash page tests / update methods

fazilqa committed:
fix: fix sent page tests / update methods

Commit:
[MHV-52918] Added unit test to check for vaPagination on last page if one threadListItem is displayed; adjusted pagination count to fit on 400+ zoom screen; adjusted test-response mock data to contain 11 threads

Commit:
added async to unit test

Commit:
[MHV-52918] Revert pagination count to maxPageLength of 7

- 7 length is the usual default recommended by the va design system; the full length of this component is displayed on screen up to 400% zoom. There may be adjustments made by the design platform team to accommodate 500% zoom in the future.

Commit:
Merge branch 'main' into 52918-Implement-V3-Pagination-in-SM

alex-morgun-1 reviewed (approved):
Approved
